analyzer/consensus/consensus.go

```diff
@@ -80,6 +82,9 @@ func NewAnalyzer(cfg *config.BlockBasedAnalyzerConfig, genesisChainContext strin
 
 // Implements BlockProcessor interface.
 func (m *processor) PreWork(ctx context.Context) error {
+	// Start background goroutine to fetch current node height.
```

Comment:
If/when this is expended to include other layers it should (as the comment on the DB table states) likely be moved into the `blockBasedAnalyzer` or maybe even into a new, `nodeStats` analyzer, to avoid duplication. 

Maybe just include a short TODO here.

Reply:
+1 to moving this out of the consensus analyzer. It has little or nothing to do with the analysis of consensus blocks.

[Here](https://github.com/oasisprotocol/nexus/blob/d937c9bc9ca31ac8d39e521b84696863fc35e55d/analyzer/evmcontractcode/evm_contract_code.go) is a recent minimal analyzer. (Aside: The code can be made even shorter with https://app.clickup.com/t/866afuw9w, but we're not working on that yet.) It should be relatively easy to adapt it to the needs of this PR.

I see now that the ticket suggests to implement this as a goroutine in the consensus analyzer, as you did. Sorry that was misleading; the ticket is outdated. Apart from the aesthetic/architectural considerations (first paragraph above), there is a recent practical consideration that makes a simple goroutine less palatable: With fast-sync and parallel analysis, we'll be running potentially multiple consensus analyzers at once. Although maybe in fast sync we'd want to disable this goroutine altogether, IDK.

Reply:
got it, moved to a separate analyzer!